### PR TITLE
Clean up the use of 'SetProfileRoot' and 'StartProfile' in ConsoleHost

### PR DESCRIPTION
**Pull request by daxian-dbw**
# PR Summary

Use `System.Runtime.ProfileOptimization.SetProfileRoot` and `System.Runtime.ProfileOptimization.StartProfile` instead of our implementation in `ClrFacade` in `ConsoleHost` for the .NET background JIT compilation optimization.

Except for this main change, this PR includes a few other minor changes:
- Add `pwsh.dll` to the crossgen assembly list, as it does get JIT'ed at startup (though the JIT time is less than 1ms)
- Remove `EarlyParse` and rename `EarlyParseHelper` to `EarlyParse`, as the former just calls the latter and does nothing else.
- Move `EarlyStartup.Init()` before `EarlyParse`, so the warm-up tasks can run earlier.
- Update the files in `tools/performance` folder: adding comments, update assembly names, update `Invoke-PerfViewPS.ps1` to make it more usable.
   - perfview log output is not written out to the console with the current `Get-Content | Out-Host` implementation. Simply write out the log content after the perfview process exits.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@daxian-dbw It seems the WIX file needs to be updated. Please have a look.

**Comment by daxian-dbw** [comment]
@adityapatwardhan Thanks! Fix has been pushed.

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs [hunk 1]
```diff
@@ -238,16 +239,17 @@ internal sealed partial class ConsoleHost
                         throw hostException;
                     }
 
+                    ProfileOptimization.StartProfile(
+                        s_theConsoleHost.LoadPSReadline()
+                            ? "StartupProfileData-Interactive"
+                            : "StartupProfileData-NonInteractive");
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs]: We use the strings in some places. Make sense to define consts?

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs [hunk 2]
```diff
@@ -31,9 +31,9 @@ public sealed class UnmanagedPSEntry
         public static int Start(string consoleFilePath, [MarshalAs(UnmanagedType.LPArray, ArraySubType = UnmanagedType.LPWStr, SizeParamIndex = 2)]string[] args, int argc)
 #pragma warning restore 1573
         {
+            System.Management.Automation.Runspaces.EarlyStartup.Init();
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs]: Style. Please insert new line.
```suggestion
            System.Management.Automation.Runspaces.EarlyStartup.Init();

```

File: tools/performance/Invoke-PerfviewPS.ps1 [hunk 3]
```diff
@@ -8,8 +8,9 @@ param(
     $ScriptBlock,
 
     $LogFileName = '.\perfview.log',
-
-    $PowerShellPath = $(Get-Command pwsh.exe).Source)
+    $PowerShellPath = $(Get-Command pwsh.exe).Source,
```
Comment (iSazonov) [inline tools/performance/Invoke-PerfviewPS.ps1]:
```suggestion
    $PowerShellPath = $(Get-Command -Name pwsh.exe).Source,
```

File: tools/performance/Invoke-PerfviewPS.ps1 [hunk 4]
```diff
@@ -8,8 +8,9 @@ param(
...
+    $PerfViewPath = $(Get-Command PerfView.exe).Source
```
Comment (iSazonov) [inline tools/performance/Invoke-PerfviewPS.ps1]:
```suggestion
    $PerfViewPath = $(Get-Command -Name PerfView.exe).Source
```

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs (hunk 1 above)
Comment (daxian-dbw) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs]: Those strings are only used once here, so I wouldn't bother to define consts for them.

**Review by iSazonov — APPROVED** [review]

**Pull request merged**